plugin-server/src/utils/periodic-task.ts

```diff
@@ -0,0 +1,57 @@
+import { instrument } from './metrics'
+import { status } from './status'
+import { sleep } from './utils'
+
+export class PeriodicTask {
+    public readonly promise: Promise<void>
+    private running = true
+    private abortController: AbortController
+
+    constructor(public name: string, task: () => Promise<void>, intervalMs: number, minimumWaitMs = 0) {
+        this.abortController = new AbortController()
+
+        const abortRequested = new Promise((resolve) => {
+            this.abortController.signal.addEventListener('abort', resolve, { once: true })
+        })
+
+        this.promise = new Promise(async (resolve, reject) => {
+            try {
+                status.debug('🔄', `${this}: Starting...`)
+                while (!this.abortController.signal.aborted) {
+                    const startTimeMs = Date.now()
+                    await instrument({ metricName: this.name }, task)
+                    const durationMs = Date.now() - startTimeMs
+                    const waitTimeMs = Math.max(intervalMs - durationMs, minimumWaitMs)
+                    status.debug(
+                        '🔄',
+                        `${this}: Task completed in ${durationMs / 1000}s, next evaluation in ${waitTimeMs / 1000}s`
```

Comment:
would it make sense to have something in VictoriaMetrics too? e.g. maybe p99, p90, p50 times

Reply:
My impression was that the `instrument` call here should be doing that: https://github.com/PostHog/posthog/blob/fb43fbde417bb80a8b7d35159b8c9af546cce17c/plugin-server/src/utils/periodic-task.ts#L22

I'm still not super confident in my understandings of our metrics setup though. Is there something else/additional I should have here?